pkg/operator/oauthapiencryption/oauthapiencryptioncontroller.go

```diff
@@ -29,102 +27,98 @@ This annotation indicates that OAS-O manages this secret.`
 	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
+	deployer statemachine.Deployer,
 	eventRecorder events.Recorder) factory.Controller {
 
 	controllerFactory := factory.New()
 	target := &oauthEncryptionConfigSyncController{
 		oauthAPIServerTargetNamespace: oauthAPIServerTargetNamespace,
-		secretLister:                  kubeInformersForNamespaces.InformersFor(operatorclient.GlobalMachineSpecifiedConfigNamespace).Core().V1().Secrets().Lister().Secrets(operatorclient.GlobalMachineSpecifiedConfigNamespace),
-		secretClient:                  secretClient.Secrets(operatorclient.GlobalMachineSpecifiedConfigNamespace),
+
+		secretListerForGlobalNamespace:      kubeInformersForNamespaces.InformersFor(operatorclient.GlobalMachineSpecifiedConfigNamespace).Core().V1().Secrets().Lister().Secrets(operatorclient.GlobalMachineSpecifiedConfigNamespace),
+		secretListerForOAuthTargetNamespace: kubeInformersForNamespaces.InformersFor(oauthAPIServerTargetNamespace).Core().V1().Secrets().Lister().Secrets(oauthAPIServerTargetNamespace),
+
+		secretClient: secretClient,
+		deployer:     deployer,
 	}
 
 	controllerFactory.WithSync(target.sync)
 	controllerFactory.WithInformers(kubeInformersForNamespaces.InformersFor(operatorclient.GlobalMachineSpecifiedConfigNamespace).Core().V1().Secrets().Informer())
 	return controllerFactory.ToController(name, eventRecorder.WithComponentSuffix("oauth-apiserver-encryption-cfg-sync-controller"))
 }
 
-// sync starts managing oauth-apiserver encryption config (encryption-config-openshift-oauth-apiserver in openshift-config-managed namespace) as described in https://github.com/openshift/enhancements/blob/master/enhancements/etcd/etcd-encryption-for-separate-oauth-apis.md
+// sync starts cleaning up oauth-apiserver encryption config (encryption-config-openshift-oauth-apiserver in openshift-config-managed namespace AND encryption-config in openshift-oauth-apiserver)
 //
-// case 1: if the secret doesn't exist and encryption is on then:
-//         - it will add the secret with the annotation (must be atomic operation) because CAO will start managing its own config iff the secret exist without the annotation
+// case 1: no-op: when the encryption config (oauth-apiserver) doesn't exist in the global namespace and the encryption is off, let CAO manage its own encryption config
 //
-// case 2: if the secret exists and it is annotated
-//         - it will simply start synchronisation
+// case 2: if the encryption config (oauth-apiserver) in the global namespace exists and it is annotated (an upgrade from 4.6)
+//         - remove the annotation
 //
-// case 3: no-op: when the secret exits but it doesn't have the annotation - that means it was created by CAO in 4.7 and this is downgrade
-// case 4: no-op: when the secret doesn't exist and encryption is off
+// case 2.1: if the encryption config (oauth-apiserver) in the target namespace exists and it is annotated (an upgrade from 4.6)
```

Comment:
"**also** exists" right? These cases are independent, i.e. we remove both annotations if they exist